### PR TITLE
[Impeller] combine text shaders.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Combine both text shaders into one. One more down!

https://github.com/flutter/flutter/issues/143540

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Version 1:

```
Changes found in shader flutter/impeller/entity/gles/glyph_atlas.frag.gles on core Mali-G78:
has_uniform_computation:
  False <- before
  True <- after
longest_path_cycles in variant Main
        arith_total arith_fma   arith_cvt   arith_sfu   load_store  varying     texture     bound       
before  0.031250    0.031250    0.031250    0.000000    0.000000    0.500000    0.250000    varying
after   0.109375    0.031250    0.109375    0.000000    0.000000    0.250000    0.250000    varying,texture

shortest_path_cycles in variant Main
        arith_total arith_fma   arith_cvt   arith_sfu   load_store  varying     texture     bound       
before  0.031250    0.031250    0.000000    0.000000    0.000000    0.500000    0.250000    varying
after   0.062500    0.031250    0.062500    0.000000    0.000000    0.250000    0.250000    varying,texture

total_cycles in variant Main
        arith_total arith_fma   arith_cvt   arith_sfu   load_store  varying     texture     bound       
before  0.031250    0.031250    0.031250    0.000000    0.000000    0.500000    0.250000    varying
after   0.125000    0.062500    0.125000    0.000000    0.000000    0.250000    0.250000    varying,texture

In variant Main:
  uniform_registers_used: 4 <- before
  uniform_registers_used: 6 <- after
```

File: impeller/entity/shaders/glyph_atlas.frag [hunk 1]
```diff
@@ -10,17 +10,31 @@ uniform f16sampler2D glyph_atlas_sampler;
...
+in highp vec2 v_uv;
 
 out f16vec4 frag_color;
 
 void main() {
   f16vec4 value = texture(glyph_atlas_sampler, v_uv);
-  if (use_alpha_color_channel == 1.0) {
-    frag_color = value.aaaa * v_text_color;
+
+  if (frag_info.is_color_glyph == 1.0) {
```
Comment (jonahwilliams) [inline impeller/entity/shaders/glyph_atlas.frag:27]: I looked at using mix(...) but malioc said it was more expensive.

File: impeller/entity/contents/text_contents.cc [hunk 2]
```diff
@@ -188,7 +185,7 @@ bool TextContents::Render(const ContentContext& renderer,
 
             for (const Point& point : unit_points) {
               vtx.unit_position = point;
-              std::memcpy(vtx_contents++, &vtx, sizeof(VS::PerVertexData));
```
Comment (jonahwilliams) [inline impeller/entity/contents/text_contents.cc:191]: I don't know why I wrote this with memcpy

**Review by chinmaygarde — APPROVED** [review]
I usually don't pay too much attention to uniform computation. Its the register spills.

**Pull request merged**